security/passwords.rst

```diff
@@ -646,6 +648,12 @@ the name of the hasher to use::
         }
     }
 
+.. note::
+
+    When migrating passwords, you don't need to implement ``PasswordHasherAwareInterface``
+    to return the legacy hasher name:
+    Symfony will detect it from your ``migrate_from`` configuration.
```

Comment:
Yeah, and if I read https://github.com/symfony/symfony/issues/48348#issuecomment-1328810098 correctly using both features together can create some confusing situations as it might by-pass the migrating hasher.

Reply:
Yes in fact you must not implement `PasswordHasherAwareInterface` **to** migrate passwords, but you can implement it *while** migrating password. That is, the two are de-correlated but you can shoot you in the foot thinking they’re not.

Do you still think we should always discourage implementing `PasswordHasherAwareInterface`?